docs/development/release.md

```diff
@@ -3,148 +3,297 @@
+
+    ```bash
+    git checkout release-1.4
+    git pull --rebase upstream release-1.4
     ```
 
-* The [image release CI pipeline](https://github.com/ray-project/kuberay/blob/master/.github/workflows/image-release.yaml) also publishes the `github.com/ray-project/kuberay/ray-operator@vX.Y.Z-rc.0` Go module. KubeRay has supported Go modules since v0.6.0. Follow these instructions to verify the Go module installation.
+2. Verify the latest commit is the version bump commit:
 
-    ```sh
-    # Install the module. This step is highly possible to fail because the module is not available in the proxy server.
-    go install github.com/ray-project/kuberay/ray-operator@v1.0.0-rc.0
+    ```bash
+    git log -1
+    # Check the commit message and changes:
+    # git show
+    ```
 
-    # Make the module available by running the go list command to prompt Go to update its index of modules with information about the module you’re publishing.
-    # See https://go.dev/doc/modules/publishing for more details.
-    GOPROXY=proxy.golang.org go list -m github.com/ray-project/kuberay/ray-operator@v1.0.0-rc.0
-    # [Expected output]: github.com/ray-project/kuberay/ray-operator v1.0.0-rc.0
+3. Create the git tag (using the `vX.Y.Z` format, e.g., `v1.4.0`):
+
+    ```bash
+    # Create a tag
+    git tag v1.4.0
+    ```
 
-    # Wait for a while until the URL https://sum.golang.org/lookup/github.com/ray-project/kuberay/ray-operator@vX.Y.Z-rc.0 no longer displays "not found". This may take 15 mins based on my experience.
-    go install github.com/ray-project/kuberay/ray-operator@v1.0.0-rc.0
+4. Push the tag to the `upstream` repository:
 
-    # Check the module is installed successfully.
-    ls $GOPATH/pkg/mod/github.com/ray-project/kuberay/
-    # [Expected output]: ray-operator@v1.0.0-rc.0
+    ```bash
+    git push upstream v1.4.0
     ```
 
-* Release rc0 Helm charts following the [instructions](../release/helm-chart.md).
+---
+
+### Step 4: Publish Release Images
+
+Trigger the [`release-image-build`](https://github.com/ray-project/kuberay/actions/workflows/image-release.yaml) workflow to build and publish the KubeRay container images.
+
+1. Navigate to the workflow page: [https://github.com/ray-project/kuberay/actions/workflows/image-release.yaml](https://github.com/ray-project/kuberay/actions/workflows/image-release.yaml)
+2. Click the **"Run workflow"** dropdown button.
+3. Set the parameters:
+    * **Use workflow from:** Select **`Tags`** and choose the tag you just pushed (e.g., **`v1.4.0`**).
+    * **Commit reference:** Leave this blank or ensure it points to the commit associated with the tag `v1.4.0`. *Usually derived from the tag.*
+    * **Desired release version:** Enter the tag name (e.g., **`v1.4.0`**). This determines the image tag.
+4. Click **"Run workflow"**.
+
+**Verification:** Monitor the workflow run. Once completed successfully, check that the corresponding image tags are available on quay.io.
```

Comment:
Add a hyperlink to https://quay.io/repository/kuberay/operator?tab=tags.

Reply:
done